code/modules/chemistry/tools/ampoules.dm

```diff
@@ -42,6 +42,47 @@
 	playsound(user.loc, "sound/impact_sounds/Generic_Snap_1.ogg", 50, 1)
 	return
 
+// based off of plants.dm
+/obj/item/reagent_containers/ampoule/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/spacecash) || istype(W, /obj/item/paper))
+		var/obj/item/clothing/mask/cigarette/custom/ampoule/C = new(user.loc)
+		C.reagents.maximum_volume = (src.reagents.total_volume)
+		src.force_drop(user)
```

Comment:
this code here seems a little weird, considering you're both dropping the ampoule and making a new cigarette.

Reply:
I had issues when I moved it into the cigarette without dropping it, so thats why its there.